ConfigData.h

```diff
@@ -122,29 +122,28 @@ class ConfigData{
         void collect_config_data(std::vector<std::string> configs = std::vector<std::string>())
         {
             std::smatch sm;
-            std::vector<std::string> data;
 
             if (configs.size() > 0)
             {
                 for (int i = 0; i < configs.size(); i++)
                 {
                     const std::string s = configs[i];
 
-                    if(std::regex_search(s.begin(), s.end(), sm, std::regex(".*=(.*)")))
-                        data.push_back(sm[1]);
-                }
-
-                if(!data.empty())
-                {
-                    this->_title       = data[0];
-                    this->_author      = data[1];
-                    this->_genre       = data[2];
-                    this->_description = data[3];
-                    this->_rating      = data[4];
-                    this->_language    = data[5];
-                    this->_image       = data[6];
-                    this->_exe         = data[7];
-                    this->_repo        = data[8];
+                    if(std::regex_search(s.begin(), s.end(), sm, std::regex("(.*)=(.*)")))
+                    {
+                        for (int i = 1; i < sm.size(); i++)
```

Comment:
Should this loop be iterated from `i=0` through `sm.size() -1`? (I may be misreading)

Reply:
index[0] of smatch is the entire string, index[1] is the first match, index[2] is the second match.
eg. 
index[0]: title=Mario
index[1]: title
index[2]: Mario

Since we don't need the entire string at all, I figured to just start it at 1